cmd/assetsvc/handler.go

```diff
@@ -336,3 +379,25 @@ func newChartVersionListResponse(c *models.Chart) apiListResponse {
 
 	return cvl
 }
+
+func decodeParam(param string, w http.ResponseWriter) string {
+	decodedParam, err := url.PathUnescape(param)
+	if err != nil {
+		log.WithError(err).Errorf("could not decode param %s", decodedParam)
+		if w != nil {
+			response.NewErrorResponse(http.StatusBadRequest, "could not decode param: "+param).Write(w)
+		}
+	}
+	return decodedParam
+}
+
+func getEncodedChartID(c *models.Chart) string {
```

Comment:
Not sure why this fn is required - it'd be clearer for call-sites to just call what you've got on the following line?

Reply:
Yep, totally right!